### DISCUSSION TITLE
OpenEBS (cstor) - discussion about usage in production environments

### DISCUSSION THREAD

**Discussion by RandomGHUser** [openebs/openebs#3370] [General] [answered]
"cStor is the recommended way to provide additional resilience to workloads via OpenEBS and is the second most widely deployed storage engine behind LocalPV."

The Openebs main page states "Used in production by Bloomberg optoro orange arista cncf comcast".  Obviously this doesn't necessarily mean they are using the cStor operators.

I've been testing out Openebs with cStor for a few weeks, as I want to be able to scale my persistent volumes easily and resiliently between k8s nodes.  But before I am able to even think about usage in a production environment, I need to ensure that there is a reliable backup solution for these PVs.  I've tried using Velero and my own automated snapshot>backup process, both have given me inconsistent results - some days working, some days giving random, unexpected errors which have taken down all my volumes on a block device (see https://github.com/openebs/openebs/issues/3368).   I am also finding that many issues I'm having are not well documented or solved through the issues page or other question sites (serverfault etc.).

cStor is listed as "beta", but with the guide stating it as being the recommended way for workloads (first line of post), I'd like to discuss with people that are actually using replicated cStor volumes with a backup solution in a production environment, what the experience has been like, have they experienced major errors or downtime, and describing their diagnosis and/or remedy process.

**Comment by avishnu** [comment, marked as answer]
Apologies for the super late response !!
CStor reached the stable state with the OpenEBS 3.0 release. Having said that, the maintainers behind OpenEBS are actively building Mayastor to be a more-than-worthy replacement for CStor. Mayastor is the currently stable and active project and CStor/Jiva are almost EOL. Recommend you to try Mayastor and share your experiences.
